### PR TITLE
feat(Core/Maps): Improvements to Cinematic function 

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/de918a0f644794beebff2234b53980b577b27e39

## Tests Performed:

- Tested
- 

## How to Test the Changes:

1. Test whatever you know about Cinematic 
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Object/Object.cpp [hunk 1]
```diff
@@ -1620,7 +1620,7 @@ float WorldObject::GetGridActivationRange() const
 {
     if (ToPlayer())
     {
-        if (ToPlayer()->IsOnCinematic())
+        if (ToPlayer()->GetCinematicMgr()->IsOnCinematic())
```
Comment (Kitzunu) [inline src/server/game/Entities/Object/Object.cpp:1623]: Why not define sCinematicMgr?

File: src/server/game/Entities/Player/Player.h [hunk 2]
```diff
@@ -2866,6 +2859,8 @@ class Player : public Unit, public GridObject<Player>
     Item* _StoreItem(uint16 pos, Item* pItem, uint32 count, bool clone, bool update);
     Item* _LoadItem(CharacterDatabaseTransaction trans, uint32 zoneId, uint32 timeDiff, Field* fields);
 
+    CinematicMgr* _cinematicMgr;
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Player/Player.h:2862]: maybe [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr)

Comment (IntelligentQuantum) [reply src/server/game/Entities/Player/Player.h:2862]: std::unique_ptr<CinematicMgr> _cinematicMgr; ?

**Comment by Lynethris** [comment]
PR work. No issues find with :

- [ X ] 10 intros races
- [ X ]  Battle of Angrathar the Wrathgate

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/game/Entities/Player/Player.h (hunk 2 above)
Comment (Winfidonarleyan) [reply src/server/game/Entities/Player/Player.h:2862]: yes

**Comment by LukasVolgger** [comment]
Thx!

**Pull request merged**